src/snapshot_vec.rs

```diff
@@ -75,13 +75,20 @@ pub trait SnapshotVecDelegate {
     fn reverse(values: &mut Vec<Self::Value>, action: Self::Undo);
 }
 
-impl<D: SnapshotVecDelegate> SnapshotVec<D> {
-    pub fn new() -> SnapshotVec<D> {
+// HACK(eddyb) manual impl avoids `Default` bound on `D`.
```

Comment:
Nit: I don't like the capital HACK -- using a manual impl isn't a *hack* -- using `derive` would be, given the way that Rust works right now, since it'd be easy but wrong.

I'd rather say "Use a manual impl to avoid unnecessary `Default` bound on `D`."

Reply:
What's wrong is the way Rust works right now, IMO.

Reply:
I agree :) it's just that leaving a comment *here* won't change it!